domainservice/src/main/kotlin/systems/ajax/malov/stockanalyzer/kafka/processor/StockPriceNotificationProcessor.kt

```diff
@@ -0,0 +1,79 @@
+package systems.ajax.malov.stockanalyzer.kafka.processor
+
+import jakarta.annotation.PostConstruct
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.bson.types.ObjectId
+import org.springframework.stereotype.Component
+import reactor.core.scheduler.Schedulers
+import reactor.kafka.receiver.KafkaReceiver
+import reactor.kafka.sender.KafkaSender
+import reactor.kafka.sender.SenderRecord
+import reactor.kotlin.core.publisher.toFlux
+import systems.ajax.malov.internalapi.KafkaTopic
+import systems.ajax.malov.internalapi.output.pubsub.stock.notification_stock_price.proto.NotificationStockPrice
+import systems.ajax.malov.internalapi.output.pubsub.stock.stock_price.proto.StockPrice
+import systems.ajax.malov.stockanalyzer.entity.MongoUserTrackedSymbol
+import systems.ajax.malov.stockanalyzer.mapper.proto.BigDecimalProtoMapper.convertBigDecimalProtoToBigDecimal
+import systems.ajax.malov.stockanalyzer.repository.UserTrackedSymbolRepository
+
+@Component
+class StockPriceNotificationProcessor(
+    private val kafkaStockPriceConsumer: KafkaReceiver<String, StockPrice>,
+    private val kafkaNotificationStockPriceSender: KafkaSender<String, NotificationStockPrice>,
+    private val userTrackedSymbolRepository: UserTrackedSymbolRepository,
+) {
+    @PostConstruct
+    fun subscribeToEvents() {
+        kafkaStockPriceConsumer.receiveAutoAck()
+            .flatMap { stockPriceConsumerRecords ->
+                stockPriceConsumerRecords.flatMap { stockPriceConsumerRecord ->
+                    createUsersNotifications(stockPriceConsumerRecord)
+                }
+            }.flatMap {
+                kafkaNotificationStockPriceSender.send(it.toFlux())
+            }
+            .subscribeOn(Schedulers.boundedElastic())
+            .subscribe()
+    }
+
+    private fun createUsersNotifications(stockPriceConsumerRecord: ConsumerRecord<String, StockPrice>) =
```

Comment:
do you need all consumer record model here? or you can use just StockPrice model?